functions/break.go

```diff
@@ -0,0 +1,27 @@
+package functions
+
+// Break returns a tuple of any elements that do not satisfy the predicate up until the first time it passes, followed
+// by the rest of the elements.
+// Can be generated on any type.
+func (s SliceType) Break(f func(ElementType) bool) (before SliceType, after SliceType) {
+	if f == nil {
+		return before, s
+	}
+
+	passed := false
+
+	for _, v := range s {
+		if passed {
```

Comment:
I should have seen this earlier because this construct happened a few times.
But you could write this (cleaner) like so (I think)

```
for _,v := range s {
      if passed || f(v) {
               after = append(after, v)
               passed = true
      } else {
                before = append(before, v)
      }
}```

Which I think makes the branching more obvious and thus more readable :)